src/connections.jl

```diff
@@ -319,12 +330,16 @@ Other encodings are not explicitly handled by this package and will probably be
 See also: [`encoding`](@ref), [`reset_encoding!`](@ref)
 """
 function set_encoding!(jl_conn::Connection, encoding::String)
-    status = libpq_c.PQsetClientEncoding(jl_conn.conn, encoding)
+    lock(jl_conn.lock) do
```

Comment:
It might be a helpful pattern to define
`lock(f, jl_conn::Connection) = lock(f, jl_conn.lock)`
and the other lock related delgates.

I've seen that pattern a few times, and I think it is a fairly solid one for coarse grained locking.

Reply:
Is that worth it? I had a hard time deciding whether or not to do that.

Reply:
Given we both considered it, I think it might be.